### PR TITLE
Key info feature base

### PR DESCRIPTION
**Pull request by dushan-madetech**
## Jira ticket URL:
https://dfedigital.atlassian.net/browse/TEVA-465
https://dfedigital.atlassian.net/browse/TEVA-469
https://dfedigital.atlassian.net/browse/TEVA-127

## Changes in this PR:
- Create clear info sections and update relevant tests
- Move school address to new sub-section called 'school location'
- New sidebar with key dates to apply

### Before

Screenshot 2020-03-05 at 15 40 42

### After

Screenshot 2020-03-05 at 16 00 51

## TODO before ready for production:
- [x] Change sidebar to have key dates to apply
- [x]  Remove left/right padding from tab sections (see prototype)
- [x] Reduce heading sizes in sections (e.g. `govuk-heading-l` to `govuk-heading-m`)
- [x] Fix grid row issue with Job Title (see Figma for how it should look, sidebar is not aligned with tabs and job title should be along the top)
- [x] Agree on finalised content/rules for days remaining to apply logic

~~Add the email instructions for flexible working vacancies (might need to get design for how that fits into new sections, see the working patterns bit of the sidebar in a part time vacancy ([e.g.](https://teaching-vacancies.service.gov.uk/jobs/teacher-of-mathematics-richard-hale-school-hertford)) to see what this looks like now, can be accessed with `vacancy.flexible_working`)~~

Agreed to get rid of this with Isabel and Chris.

~~Fix back link (the one at the top of a vacancy) so it works with tabs. It should take you back to the list of vacancies but if you've clicked on tabs it goes to the last clicked tab)~~

Created [a ticket](https://dfedigital.atlassian.net/browse/TEVA-570) to fix this bug

- [x] Rebase from develop once https://github.com/DFE-Digital/teacher-vacancy-service/pull/1335 has been merged in to develop 
- [x] Solve any merge conflicts
- [x] Remove any code which might be duplicated (e.g. presenter methods for school type)
- [x] Add/amend any keys/values that need to be added/amended in the `en.yml` file
- [x] Add tests to the presenter methods that have been added and don't have tests
- [x] Refactor code and improve it where possible 
- [x] Address comments on this PR
- [x] Request another review since new code has been added since the first set of reviews and address any comments as necessary
- [x] Check the hiring staff job creation journey to see the new names of fields (might have been affected by changes to the en.yml)
- [ ] Go over how this looks with different vacancies on a non production environment (preferably with real school data)

### CODE REVIEW

File: app/frontend/styles/application.scss [hunk 1]
```diff
@@ -91,3 +91,20 @@ p {
...
+}
+
+a.govuk-tabs__tab.tab-links-override:link,
+.govuk-tabs__tab.tab-links-override:visited,
+a.js-enabled .govuk-tabs__tab.tab-links-override:link,
+a.js-enabled .govuk-tabs__tab.tab-links-override:visited {
+    color: $govuk-link-colour;
+}
+
+li.govuk-tabs__list-item--selected > a.govuk-tabs__tab.govuk-tabs__tab.tab-links-override,
```
Comment (david-mears-dfe) [inline app/frontend/styles/application.scss]: Could we separate these concerns into the most specific file e.g. `app/styles/components/vacancies`?

File: app/views/hiring_staff/vacancies/_show.html.haml [hunk 2]
```diff
@@ -35,60 +35,21 @@
...
-      %p= @vacancy.job_description
+      .govuk-tabs{"data-module": "govuk-tabs"}
+        %h2.govuk-tabs__title
+          = t('jobs.key_information')
+        %ul.govuk-tabs__list
+          %li.govuk-tabs__list-item.govuk-tabs__list-item--selected
+            %a.govuk-tabs__tab.tab-links-override{:href => "#job-details"}
+              = t('jobs.job_details')
+          %li.govuk-tabs__list-item
+            %a.govuk-tabs__tab.tab-links-override{:href => "#school-overview"}
```
Comment (david-mears-dfe) [inline app/views/hiring_staff/vacancies/_show.html.haml]: Does rubocop alert you `Style/HashSyntax: Use the new Ruby 1.9 hash syntax`? Not sure if we have linting on these haml files.

Comment (dushan-madetech) [reply app/views/hiring_staff/vacancies/_show.html.haml]: Nope it didn't, but I can change this file to use the newer syntax 😄

File: app/frontend/styles/application.scss (hunk 1 above)
Comment (dushan-madetech) [reply app/frontend/styles/application.scss]: Yep, will get that done

File: app/presenters/school_presenter.rb [hunk 3]
```diff
@@ -3,6 +3,20 @@ def location
     [model.name, model.town, model.county].reject(&:blank?).join(', ')
   end
 
+  def school_size
+    if model.gias_data.present?
+      return model.gias_data['NumberOfPupils'] + ' pupils enrolled' if model.gias_data['NumberOfPupils'].present?
```
Comment (tatyree) [inline app/presenters/school_presenter.rb]: I accept it's an edge case, but you should probably use `pluralize` here for those one-pupil schools :)

Comment (dushan-madetech) [reply app/presenters/school_presenter.rb]: I didn't even consider that edge case 😅 but that's a good point. I'll make sure to amend 😄

File: app/presenters/school_presenter.rb [hunk 4]
```diff
@@ -3,6 +3,20 @@ def location
...
+      return 'Up to ' + model.gias_data['SchoolCapacity'] + 'pupils' if model.gias_data['SchoolCapacity'].present?
+    end
+    'No data available'
+  end
+
+  def ofsted_report
+    if model.gias_data.present?
+      "https://www.ofsted.gov.uk/oxedu_providers/full/(urn)/#{model.gias_data['URN']}"
```
Comment (tatyree) [inline app/presenters/school_presenter.rb]: Make the ofstead endpoint a frozen constant at the top of the presenter, please.  Makes is easier to find and change if we have to.

File: app/presenters/school_presenter.rb [hunk 5]
```diff
@@ -3,6 +3,20 @@ def location
     [model.name, model.town, model.county].reject(&:blank?).join(', ')
   end
 
+  def school_size
+    if model.gias_data.present?
+      return model.gias_data['NumberOfPupils'] + ' pupils enrolled' if model.gias_data['NumberOfPupils'].present?
+      return 'Up to ' + model.gias_data['SchoolCapacity'] + 'pupils' if model.gias_data['SchoolCapacity'].present?
+    end
+    'No data available'
```
Comment (david-mears-dfe) [inline app/presenters/school_presenter.rb]: Should this 'No data available' come from en.yml?

File: app/presenters/school_presenter.rb (hunk 4 above)
Comment (dushan-madetech) [reply app/presenters/school_presenter.rb]: Will do! Do you have a name you'd prefer? I'll probably go with `OFSTED_REPORT_ENDPOINT` unless you have a preference for a different name

File: app/presenters/school_presenter.rb (hunk 5 above)
Comment (dushan-madetech) [reply app/presenters/school_presenter.rb]: Yep! Good catch 😄

Comment (cpjmcquillan) [reply app/presenters/school_presenter.rb]: Can we use string interpolation here?

File: app/views/vacancies/_job_details_section.html.haml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+          - if @vacancy.other_subjects.nil? || @vacancy.other_subjects == ''
+            = @vacancy.main_subject
+          - else 
+            = @vacancy.main_subject + ', ' + @vacancy.other_subjects
+      %tr.govuk-table__row
+        %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" } 
+          = t('jobs.working_patterns')
+        %td.govuk-table__cell
+          = @vacancy.working_patterns
+          -# - if @vacancy.flexible_working?
```
Comment (tatyree) [inline app/views/vacancies/_job_details_section.html.haml]: Unless you're putting it in as an example for a TODO, like you do below, just take commented out code out, please.

File: app/presenters/school_presenter.rb [hunk 7]
```diff
@@ -20,4 +38,18 @@ def age_range
       I18n.t('schools.not_given')
     end
   end
+
+  private
+
+  def number_of_pupils
+    [model.gias_data['NumberOfPupils'].to_s, pupils, I18n.t('schools.size.enrolled')].join(' ')
```
[the remark this answers is not in the document]
Comment (david-mears-dfe) [reply app/presenters/school_presenter.rb]: That's surprising - they're there on line 434.

File: app/presenters/school_presenter.rb [hunk 8]
```diff
@@ -9,6 +13,20 @@ def location
     [model.name, model.town, model.county].reject(&:blank?).join(', ')
   end
 
+  def school_size
+    if model.gias_data.present?
+      return number_of_pupils if model.gias_data['NumberOfPupils'].present?
```
[the remark this answers is not in the document]
Comment (david-mears-dfe) [reply app/presenters/school_presenter.rb]: Very interested in the cleaner method. I broke it out into methods in an attempt to appease Rubocop's iron linter.

File: app/views/hiring_staff/schools/show.html.haml [hunk 9]
```diff
@@ -28,7 +28,7 @@
     %table.check-your-answers.govuk-table
       %thead.govuk-table__head
         %tr.govuk-table__row
-          %th.govuk-table__header{:colspan => "2"}
+          %th.govuk-table__header{ :colspan => "2" }
```
[the remark this answers is not in the document]
Comment (david-mears-dfe) [reply app/views/hiring_staff/schools/show.html.haml]: I'm not sure that rubocop reads the haml. I just installed the haml-lint gem and got `108 files inspected, 1507 lints detected`. I'll make a ticket for getting the haml files linted.

File: app/frontend/styles/application.scss [hunk 10]
```diff
@@ -18,32 +18,43 @@ $govuk-image-url-function: frontend-image-url;
 
 @import 'components/awaiting_feedback_vacancies';
 @import 'components/banners';
+@import 'components/banners';
```
Comment (cesidio) [inline app/frontend/styles/application.scss]: We have duplication of file imports in this file

File: app/frontend/styles/components/vacancies-show.scss [hunk 11]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (cesidio) [inline app/frontend/styles/components/vacancies-show.scss]: Empty line at the start of the file

**Pull request closed without merging**